### DISCUSSION TITLE
如何编译x86的版本

### DISCUSSION THREAD

**Discussion by carbon1985** [Tencent/libpag#1734] [Q&A] [answered]
如何在mac上构建x86的动态库？我看源码里面，只有编译arm架构的so，我需要编译x86架构的so，请问有什么办法可以实现呢？

我尝试修改abiFilters "x86_64"，然后构建时出现下面的错误：
  FAILED: tgfx/libtgfx.a 
  : && /Users/terrytan/Library/Android/sdk/cmake/3.10.2.4988404/bin/cmake -E remove tgfx/libtgfx.a && /Users/terrytan/Library/Android/sdk/ndk/19.2.5345600/toolchains/llvm/prebuilt/darwin-x86_64/bin/x86_64-linux-android-ar qc tgfx/libtgfx.a  tgfx/CMakeFiles/tgfx.dir/src/core/Bitmap.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/Canvas.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/Color.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/Data.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/EncodeOrigin.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/Font.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/Image.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/ImageBuffer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/ImageCodec.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/ImageFilter.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/ImageInfo.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/ImageReader.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/ImageStream.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/Mask.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/Matrix.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/Paint.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/Path.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/PathEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/PathMeasure.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/PathRef.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/PixelBuffer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/PixelRef.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/PixelRefMask.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/Pixmap.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/Rect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/Shape.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/SimpleTextBlob.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/YUVColorSpace.cpp.o tgfx/CMakeFiles/tgfx.dir/src/core/YUVData.cpp.o tgfx/CMakeFiles/tgfx.dir/src/effects/BlurImageFilter.cpp.o tgfx/CMakeFiles/tgfx.dir/src/effects/ColorMatrixFilter.cpp.o tgfx/CMakeFiles/tgfx.dir/src/effects/DropShadowImageFilter.cpp.o tgfx/CMakeFiles/tgfx.dir/src/effects/LumaColorFilter.cpp.o tgfx/CMakeFiles/tgfx.dir/src/effects/ModeColorFilter.cpp.o tgfx/CMakeFiles/tgfx.dir/src/effects/ShaderMaskFilter.cpp.o tgfx/CMakeFiles/tgfx.dir/src/images/AsyncSource.cpp.o tgfx/CMakeFiles/tgfx.dir/src/images/BufferSource.cpp.o tgfx/CMakeFiles/tgfx.dir/src/images/EncodedSource.cpp.o tgfx/CMakeFiles/tgfx.dir/src/images/ImageGeneratorTask.cpp.o tgfx/CMakeFiles/tgfx.dir/src/images/ImageSource.cpp.o tgfx/CMakeFiles/tgfx.dir/src/images/RGBAAAImage.cpp.o tgfx/CMakeFiles/tgfx.dir/src/images/RasterBuffer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/images/RasterGenerator.cpp.o tgfx/CMakeFiles/tgfx.dir/src/images/RasterYUVBuffer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/images/SubsetImage.cpp.o tgfx/CMakeFiles/tgfx.dir/src/images/TextureSource.cpp.o tgfx/CMakeFiles/tgfx.dir/src/shaders/ColorFilterShader.cpp.o tgfx/CMakeFiles/tgfx.dir/src/shaders/ColorShader.cpp.o tgfx/CMakeFiles/tgfx.dir/src/shaders/GradientShader.cpp.o tgfx/CMakeFiles/tgfx.dir/src/shaders/ImageShader.cpp.o tgfx/CMakeFiles/tgfx.dir/src/shaders/LocalMatrixShader.cpp.o tgfx/CMakeFiles/tgfx.dir/src/shapes/PathProxy.cpp.o tgfx/CMakeFiles/tgfx.dir/src/shapes/PathShape.cpp.o tgfx/CMakeFiles/tgfx.dir/src/shapes/RRectShape.cpp.o tgfx/CMakeFiles/tgfx.dir/src/shapes/RectShape.cpp.o tgfx/CMakeFiles/tgfx.dir/src/shapes/TextureShape.cpp.o tgfx/CMakeFiles/tgfx.dir/src/shapes/TriangulatingShape.cpp.o tgfx/CMakeFiles/tgfx.dir/src/utils/Buffer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/utils/BytesKey.cpp.o tgfx/CMakeFiles/tgfx.dir/src/utils/Clock.cpp.o tgfx/CMakeFiles/tgfx.dir/src/utils/DataView.cpp.o tgfx/CMakeFiles/tgfx.dir/src/utils/PixelFormatUtil.cpp.o tgfx/CMakeFiles/tgfx.dir/src/utils/Stream.cpp.o tgfx/CMakeFiles/tgfx.dir/src/utils/Task.cpp.o tgfx/CMakeFiles/tgfx.dir/src/utils/TaskGroup.cpp.o tgfx/CMakeFiles/tgfx.dir/src/utils/UTF.cpp.o tgfx/CMakeFiles/tgfx.dir/src/utils/UniqueID.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/AARectEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/Backend.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/Blend.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ColorMatrixFragmentProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ConstColorProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/Context.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/CoordTransform.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/DefaultGeometryProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/Device.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/DeviceSpaceTextureEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/DrawingManager.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/DualBlurFragmentProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/EllipseGeometryProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/EmptyXferProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ExternalTexture.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/FragmentProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/FragmentShaderBuilder.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/GLFragmentProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/GLGeometryProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/GeometryProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/Gpu.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/GradientCache.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/LumaColorFilterEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/OpsRenderPass.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/Pipeline.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/PlainTexture.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/PorterDuffXferProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ProgramBuilder.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ProgramCache.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ProxyProvider.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/Quad.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/QuadPerEdgeAAGeometryProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/Resource.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ResourceCache.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ResourceProvider.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/SamplerState.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/SeriesFragmentProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ShaderBlend.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ShaderBuilder.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/Surface.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/SurfaceDrawContext.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/Texture.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/TextureEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/TextureProxy.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/TextureResolveRenderTask.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/TiledTextureEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/UniqueKey.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/VaryingHandler.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/VertexShaderBuilder.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/Window.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/XfermodeFragmentProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/YUVTexture.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/YUVTextureEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/gradients/ClampedGradientEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/gradients/DualIntervalGradientColorizer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/gradients/LinearGradientLayout.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/gradients/RadialGradientLayout.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/gradients/SingleIntervalGradientColorizer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/gradients/SweepGradientLayout.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/gradients/TextureGradientColorizer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/gradients/UnrolledBinaryGradientColorizer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ops/ClearOp.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ops/DrawOp.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ops/FillRectOp.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ops/Op.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ops/OpsTask.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ops/RRectOp.cpp.o tgfx/CMakeFiles/tgfx.dir/src/gpu/ops/TriangulatingPathOp.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/Print.cpp.o tgfx/CMakeFiles/tgfx.dir/src/codecs/png/PngCodec.cpp.o tgfx/CMakeFiles/tgfx.dir/src/vectors/freetype/FTFace.cpp.o tgfx/CMakeFiles/tgfx.dir/src/vectors/freetype/FTMask.cpp.o tgfx/CMakeFiles/tgfx.dir/src/vectors/freetype/FTPath.cpp.o tgfx/CMakeFiles/tgfx.dir/src/vectors/freetype/FTScalerContext.cpp.o tgfx/CMakeFiles/tgfx.dir/src/vectors/freetype/FTTypeface.cpp.o tgfx/CMakeFiles/tgfx.dir/src/vectors/freetype/FTUtil.cpp.o tgfx/CMakeFiles/tgfx.dir/src/vectors/freetype/SystemFont.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLAARectEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLAssembledGLESInterface.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLAssembledGLInterface.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLAssembledWebGLInterface.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLBlend.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLBuffer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLCaps.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLClampedGradientEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLColorMatrixFragmentProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLConstColorProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLContext.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLDefaultGeometryProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLDevice.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLDeviceSpaceTextureEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLDualBlurFragmentProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLDualIntervalGradientColorizer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLEllipseGeometryProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLEmptyXferProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLFragmentShaderBuilder.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLFunctions.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLGpu.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLInterface.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLLinearGradientLayout.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLLumaColorFilterEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLOpsRenderPass.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLPorterDuffXferProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLProgram.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLProgramBuilder.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLProgramCreator.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLProgramDataManager.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLQuadPerEdgeAAGeometryProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLRadialGradientLayout.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLRenderTarget.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLSampler.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLSemaphore.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLSeriesFragmentProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLSingleIntervalGradientColorizer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLSweepGradientLayout.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLTextureEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLTextureGradientColorizer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLTiledTextureEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLUniformHandler.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLUnrolledBinaryGradientColorizer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLUtil.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLVertexShaderBuilder.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLXfermodeFragmentProcessor.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/GLYUVTextureEffect.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/android/AHardwareBufferFunctions.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/android/AndroidBitmap.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/android/GLExternalOESTexture.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/android/HandlerThread.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/android/HardwareBuffer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/android/JNIEnvironment.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/android/JNIInit.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/android/JNIUtil.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/android/NativeCodec.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/android/NativeImageBuffer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/android/Print.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/android/SurfaceTexture.cpp.o tgfx/CMakeFiles/tgfx.dir/src/platform/android/SurfaceTextureReader.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/egl/EGLDevice.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/egl/EGLDisplayWrapper.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/egl/EGLGlobals.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/egl/EGLHardwareBuffer.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/egl/EGLHardwareTexture.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/egl/EGLProcGetter.cpp.o tgfx/CMakeFiles/tgfx.dir/src/opengl/egl/EGLWindow.cpp.o && /Users/terrytan/Library/Android/sdk/ndk/19.2.5345600/toolchains/llvm/prebuilt/darwin-x86_64/bin/x86_64-linux-android-ranlib tgfx/libtgfx.a && cd /Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx && node /Users/terrytan/libpag/vendor_tools/merge -p android -a x64 -v /Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx/libtgfx.a /Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx/CMakeFiles/tgfx-vendor.dir/x64/libtgfx-vendor.a -o /Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx/libtgfx.a
  cd /Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx/temp
  cd /Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx/temp/libtgfx
  /Users/terrytan/Library/Android/sdk/ndk/19.2.5345600/toolchains/llvm/prebuilt/darwin-x86_64/bin/arm-linux-androideabi-ar x /Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx/libtgfx.a
  /Users/terrytan/Library/Android/sdk/ndk/19.2.5345600/toolchains/llvm/prebuilt/darwin-x86_64/bin/arm-linux-androideabi-ar t /Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx/libtgfx.a > _objectFileList
  /Users/terrytan/Library/Android/sdk/ndk/19.2.5345600/toolchains/llvm/prebuilt/darwin-x86_64/bin/arm-linux-androideabi-ar xN 1 /Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx/libtgfx.a Print.cpp.o && mv Print.cpp.o 1.Print.cpp.o
  /Users/terrytan/Library/Android/sdk/ndk/19.2.5345600/toolchains/llvm/prebuilt/darwin-x86_64/bin/arm-linux-androideabi-ar xN 2 /Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx/libtgfx.a Print.cpp.o && mv Print.cpp.o 2.Print.cpp.o
  cd /Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx/temp/libtgfx-vendor
  /Users/terrytan/Library/Android/sdk/ndk/19.2.5345600/toolchains/llvm/prebuilt/darwin-x86_64/bin/arm-linux-androideabi-ar x /Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx/CMakeFiles/tgfx-vendor.dir/x64/libtgfx-vendor.a
  /Users/terrytan/Library/Android/sdk/ndk/19.2.5345600/toolchains/llvm/prebuilt/darwin-x86_64/bin/arm-linux-androideabi-ar: /Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx/CMakeFiles/tgfx-vendor.dir/x64/libtgfx-vendor.a: No such file or directory
  ninja: build stopped: subcommand failed.

/Users/terrytan/libpag/android/libpag/.cxx/Debug/2f1a2g4e/x86_64/tgfx/CMakeFiles/tgfx-vendor.dir/ 这个目录下没有x64的文件，只有arm和arm64的文件

## 【版本信息】

4.2.88

## 【平台信息】

Android 原生

**Comment by domchen** [comment]
PAG依赖一些第三方库，目前第三方库的构建工具里没有支持Android端的x86架构编译。但实际上你编译arm架构是直接可以用在x86架构的设备上的。

**Comment by hdjjun** [comment]
基于libpag，专门针对android适配了x86_64,详见https://github.com/hdjjun/libpag-android/

**Reply by ningxingxing** [reply]
使用https://github.com/hdjjun/libpag-android/   适配x86的会出现UnsatisfiedLinkError异常，无法正常使用  Caused by: java.lang.UnsatisfiedLinkError: No implementation found for void org.libpag.PAGAnimator.nativeInit() (tried Java_org_libpag_PAGAnimator_nativeInit and Java_org_libpag_PAGAnimator_nativeInit__)
                                                                                                    	at org.libpag.PAGAnimator.nativeInit(Native Method)

**Comment by domchen** [comment]
问一下大家，你们使用安卓×86架构的场景具体是什么？另外是只需要64位的还是也有32位×86架构的需求？如果需求比较普遍我们可以考虑原生支持一下。

**Reply by ningxingxing** [reply]
x86_64的，车机平台

**Reply by domchen** [reply]
了解，那我们下个版本支持一下。

**Comment by domchen** [comment, marked as answer]
#1962 已经合入主干，main 分支最新代码现在已经可以直接编译出 x86_64 架构，4.3的下个预编译版本也会带出 x86_64架构的Android AAR。
